### PR TITLE
auth: a new error code for the case of password auth against no passw…

### PR DESCRIPTION
**Pull request by mitake**
…ord user

It will be useful for auditing logs and detecting authentication request with password against a no password user, which only accepts CN based authentication.

cc @xiang90 @gyuho @spzala

### CODE REVIEW

**Comment by mitake** [comment]
I'll add a test case for hitting the new path.

File: auth/store.go [hunk 1]
```diff
@@ -333,7 +334,7 @@ func (as *authStore) CheckPassword(username, password string) (uint64, error) {
 		}
 
 		if user.Options != nil && user.Options.NoPassword {
-			return 0, ErrAuthFailed
+			return 0, ErrUserIsNoPassword
```
Comment (spzala) [inline auth/store.go:337]: @mitake cool, lgtm but if you update PR, I would suggest to change the err name to `ErrNoPasswordUser` or `ErrUserWithNoPassword` if you like it. Also, I guess in the future we can a test case if that's possible. Thanks!

Comment (mitake) [reply auth/store.go]: Got it, I used `ErrNoPasswordUser`. And I agree with adding a test case in future as an individual PR. Sorry I couldn't take a time enough.

**Comment by spzala** [comment]
Thanks @mitake lgtm

**Pull request merged**